text/0187-data-classification.md

```diff
@@ -0,0 +1,323 @@
+  return value ^ classification
+}
+```
+
+The SDK would need to add the following methods to the modified types (mentioned earlier) in order to support classification
+
+```go
+func (t *T) AppendAttribute(attr Attribute) {
+  t.attributes = append(t.attributes, attr)
+  t.Classification = Combine(t.Classification, attr.GetClassification()
+}
+
+func (t T) GetClassification() Hint {
+  return t.Classification
+}
+
+func (t *T) SetClassification(c Classification) {
+  t.Classification = c
+}
+```
+
+Once the SDK has been updated to support the definitions, along with the internal model for the collector, an extension can be used to wrap existing exporters to act on the data:
+
+```go
+type ClassifierFilterExtension struct {
+  MatchOn Classification
+  Action  func(t T) T
+}
+
+func (cfe *ClassifierFilterExtension) Match(list []T) (filtered []T) {
+  for _, t := range list {
+    if Contains(l.GetClassification(), cfe.MathchOn) {
+      t = cfe.Action(t)
+    }
+    filtered.Append(t)
+  }
+}
+
+// RemoveMathcedAttribute is an example of an Action that 
+// can be set when using classification filtering extension
+func RemoveMatchedAttributes(c Classification) func(t T) T {
+  return func(t T) T { 
+      attrs []Attribute
+      for _, attr := range t.Attributes() {
+        if !Match(attr.GetClassification(), c) {
+          attrs.Append(attr)
+        }
+      }
+      t.SetAttributes(attrs)
+      t.SetClassification(Remove(t.GetClassification(), c))
+      return t  
+  }
+}
+```
+
+## Trade-offs and mitigations
+
+- Going forward with this does imply a lot of ownership on the USER developing the instrumentation code, and a lot of potential cognitive load perceived with it.
+  - This only works with USERs implementing their code using OTEL instrumentation.
+- Converting from non OTLP formats to OTLP would need depend on processors / converters to support this, however doing so will likely impact performance.
+- Managing classifications within the semantic convention has not be explored
+- The tolerance of what is considered a given classification could change between organisations
+
+## Prior art and alternatives
+
+There was a conversation that was related here on attributes that should be subject to data regulations Proposal: Add Sensitive Data LabelsOPEN , however, this approach would superseded what is mentioned there and be applied more broadly to the project.
+
+The alternatives that were considered are:
+
+- The Semantic Convention defines classification for known attributes
+  - this was no considered heavily since it misses USERS (developers) whom are using the SDK natively and then would need to manage their own Semantic Convention that would also need to be understood by the down stream processors and vendors
+    - Does not account for version drift with a breaking change involved
+    - Does a processor need to know all versions?
+    - Does that mean values can not be changed in the semantic convention?
```

Comment:
Another downside of doing it in semantic conventions goes to your point above that different organizations will have potentially different classifications. Reaching a common agreement on classification for the semantic conventions would require lengthy discussion, with a good chance of many attributes not reaching a consensus decision on a classification